.azure-pipelines/templates/jobs/packaging-jobs.yml

```diff
@@ -59,3 +58,45 @@ jobs:
           set PATH=%ProgramFiles(x86)%\Certbot\bin;%PATH%
           venv\Scripts\python -m pytest certbot-ci\certbot_integration_tests\certbot_tests -n 4
         displayName: Run certbot integration tests
+  - job: snap_build
+    strategy:
+      matrix:
+        amd64:
+          ARCH: amd64
+        arm64:
+          ARCH: arm64
+        armhf:
+          ARCH: armhf
+    pool:
+      vmImage: ubuntu-18.04
+    steps:
+      - script: |
+          snap/local/build.sh ${ARCH}
+          mv *.snap $(Build.ArtifactStagingDirectory)
+        displayName: Build Certbot snap
+      - task: PublishPipelineArtifact@1
+        inputs:
+          path: $(Build.ArtifactStagingDirectory)
+          artifact: snap-$(arch)
+        displayName: Store snap artifact
+  - job: snap_run
+    dependsOn: snap_build
+    pool:
+      vmImage: ubuntu-18.04
+    steps:
+      - script: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends nginx-light snap
+          python tools/pip_install.py -U tox
+        displayName: Install dependencies
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          artifact: snap-amd64
+          path: $(Build.SourcesDirectory)/snap
+        displayName: Retrieve Certbot snap
+      - script: |
+          sudo snap install --dangerous --classic snap/*.snap
+        displayName: Install Certbot snap
+      - script: |
+          python -m tox -e integration-external,apacheconftest-external-with-pebble
```

Comment:
It was my initial intention, but I think it does not worth it finally. Except the `tox` call itself, operations in `snap_run` are entirely specific to snap jobs, so adding them to `tox-steps.yml` would need to have a condition of each of these steps, making `tox-steps.yml` way harder to read. We would not save lines or reusability here, but would add just a bunch of conditions.

Reply:
To be clear, what I was imagining setting `TOXENV=integration-external,apacheconftest-external-with-pebble` for the entire job and changing its steps to something like:
```
   steps:
      - task: DownloadPipelineArtifact@2
        inputs:
          artifact: snap-amd64
          path: $(Build.SourcesDirectory)/snap
        displayName: Retrieve Certbot snap
      - template: ../steps/tox-steps.yml
```
The problem with this is if we want to continue to rely on the explicit `snapd` dependency, we need some way to run `snap install --dangerous --classic snap/*.snap` after `snapd` has been installed but before running `tox`.

I think there are ways to resolve this, but the downsides may outweigh the benefit. Do whatever you want with this comment.

Reply:
There is another problem also: you cannot use `tox-steps.yml` as a template inside one specific step, since `tox-steps.yml` is in fact projecting to `steps`. It is not the correct level. Well, I really think it does not worth a refactoring here. I expect an easier merged logic once we will have builds for snap DNS plugins, their tests and their deployments.